Sources/StreamChat/Query/Filter.swift

```diff
@@ -99,6 +99,8 @@ public struct Filter<Scope: FilterScope> {
     /// The "right-hand" side of the filter. Specifies the value the filter should match.
     public let value: FilterValue
 
+    public let keyPathValueProvider: (() -> String)?
```

Comment:
What is the reason for this to be a closure? Why isn't it just a `String`? 

Also, I think we should make this internal for now and not public since the goal is only for us to provide this automatic behaviour. It will be easy for customers to commit mistakes, especially because they don't have access to the DTOs. 

Another thing that seems to be missing, is the option to override a value. In some cases, in server, we have booleans like "isModerator", but in our data, it translates to an enum. 

This is why the original implementation was like this:
https://github.com/GetStream/stream-chat-swift/blob/36e4e8e5679832238bba0f271185125c8179606f/Sources/StreamChat/Query/Filter.swift#L187-L190

We had a payloadKey, a dtoKey, and a possibility to override the value. Since a payloadKey is a CodingKey it also provided some safety, and the dtoKey would always use the #keyPath to avoid mistakes, especially if we change the DTO models in the future.

Reply:
> Another thing that seems to be missing, is the option to override a value. In some cases, in server, we have booleans like "isModerator", but in our data, it translates to an enum
I'm not really sure what you mean with this one. Let's catch up tomorrow about it